backend/Collectors/UniqueCollector.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Collectors;
+
+use Internships\Models\CollectedContent;
+use JsonSerializable;
+
+abstract class UniqueCollector implements JsonSerializable
+{
+    /** @var CollectedContent[] */
+    protected array $collectedContent;
+    protected int $nextIdToAssign;
+
+    public function __construct()
+    {
+        $this->clearData();
+    }
+
+    public function clearData(): void
+    {
+        $this->nextIdToAssign = 0;
+        $this->collectedContent = [];
+    }
+
+    public function collectAndGetID(string $content): int
+    {
+        foreach ($this->collectedContent as $collected) {
+            if ($collected->getCollectedName() === $content) {
+                return $collected->getID();
+            }
+        }
+        $newId = $this->nextIdToAssign++;
+        array_push($this->collectedContent, new CollectedContent($newId, $content));
```

Comment:
Just for your knowledge, you can also do this in this way:

```suggestion
        $this->collectedContent[] = new CollectedContent($newId, $content);
```

It's not a suggestion to change it. Just FYI.

Reply:
Hm, I'm not sure I enjoy this form of pushing into array.

I look at things more visually and at first glance I thought the second version is creating new empty array with one element. I prefer to immediately be aware what's happening, and "array_push" tells me exactly what I'm doing here.

Just a personal preference, though.